knudev-security/src/main/java/ua/knu/knudev/knudevsecurity/service/AuthenticationService.java

```diff
@@ -21,19 +25,36 @@ public class AuthenticationService implements AuthServiceApi {
 
     @Override
     public AuthenticationResponse authenticate(AuthenticationRequest authReq) {
-        AccountAuth account = accountAuthMapper.toDomain(accountService.findByEmail(authReq.email()));
+        AccountAuthDto account = accountService.findByEmail(authReq.email());
 
-        accountService.checkAccountValidity(account, account.getEmail());
+        checkAccountValidity(account, authReq.email());
 
         authenticationManager.authenticate(
                 new UsernamePasswordAuthenticationToken(authReq.email(), authReq.password())
         );
 
-        Tokens tokens = jwtService.generateTokens(account);
+        Tokens tokens = jwtService.generateTokens(accountAuthMapper.toDomain(account));
         return AuthenticationResponse.builder()
                 .accessToken(tokens.accessToken())
                 .refreshToken(tokens.refreshToken())
                 .build();
     }
 
+    private void checkAccountValidity(AccountAuthDto account, String email) throws AuthenticationException {
+        if (account == null) {
+            String accDoesNotExistErrorMsg = String.format("Account with email %s does not exist", email);
+            throw new UsernameNotFoundException(accDoesNotExistErrorMsg);
+        }
+
+        if (!account.enabled()) {
+            String disabledAccMsg = "Your account is disabled. Please activate it via link on email: " + account.email();
```

Comment:
you don`t need here this field write text in method body

Reply:
done